### PR TITLE
Upgrading components for PHP analysis (Composer 2 / PHP 8)

### PR DESCRIPTION
**Pull request by camillem**
Hi,
This PR is motivated by https://github.com/oss-review-toolkit/ort/issues/4971
I have upgraded both Composer and PHP, as it seemed relevant to real-life contexts.

Cheers,
Camille

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -93,6 +94,7 @@ RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/
         netbase \
         openssh-client \
         unzip \
+        wget \
```
Comment (sschuberth) [inline Dockerfile:98]: We already have `curl`, so please use that instead.

File: Dockerfile [hunk 2]
```diff
@@ -150,7 +152,11 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
     yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS --sdk_root=$ANDROID_HOME "platform-tools" && \
     chmod -R o+w $ANDROID_HOME && \
     # Install 'CocoaPods'.
-    gem install cocoapods -v $COCOAPODS_VERSION
+    gem install cocoapods -v $COCOAPODS_VERSION && \
+    # Install Composer V2 manually
```
Comment (sschuberth) [inline Dockerfile:159]: Let's say "Install Composer version 2 manually." (Note the dot at the end of the sentence.)

File: Dockerfile [hunk 3]
```diff
@@ -104,7 +105,9 @@ RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/
         subversion \
         # Install package managers (in versions known to work).
         cargo=$CARGO_VERSION \
-        php-cli \
+        php8.1-cli \
+        php8.1-xml \
+	php8.1-sqlite3 \
```
Comment (sschuberth) [inline Dockerfile:110]: Wrong indentation.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (camillem) [reply Dockerfile]: Corrected.

File: Dockerfile [hunk 4]
```diff
@@ -150,7 +152,11 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
     yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS --sdk_root=$ANDROID_HOME "platform-tools" && \
     chmod -R o+w $ANDROID_HOME && \
     # Install 'CocoaPods'.
-    gem install cocoapods -v $COCOAPODS_VERSION
+    gem install cocoapods -v $COCOAPODS_VERSION && \
+    # Install Composer V2 manually
+    wget https://raw.githubusercontent.com/composer/getcomposer.org/$COMPOSER_HASH/web/installer -O - -q | php -- --quiet && \
+    php -r "unlink('composer-setup.php');" && \
```
[the remark this answers is not in the document]
Comment (camillem) [reply Dockerfile]: Indeed. Removed.

File: Dockerfile [hunk 5]
```diff
@@ -150,7 +152,11 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
...
+    mv composer.phar /usr/local/bin/composer
```
[the remark this answers is not in the document]
Comment (camillem) [reply Dockerfile:160]: Yes, the installer creates `composer.phar` in the current directory.

File: Dockerfile [hunk 6]
```diff
@@ -80,6 +80,7 @@ RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/
     curl -ksS "https://keyserver.ubuntu.com/pks/lookup?op=get&options=mr&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | apt-key adv --import - && \
     curl -sL https://deb.nodesource.com/setup_16.x | bash - && \
     add-apt-repository -y ppa:git-core/ppa && \
+    add-apt-repository ppa:ondrej/php && \
```
[the remark this answers is not in the document]
Comment (camillem) [reply Dockerfile]: Added `-y` and corrected commit message.

File: Dockerfile (hunk 3 above)
Comment (camillem) [reply Dockerfile]: Corrected.

**Comment by sschuberth** [comment]
> This should be removed when upgrading base image to 22.04 LTS

This comment makes me wonder whether there are any blockers to move to Ubuntu 22.04 LTS instead?

**Comment by sschuberth** [comment]
> This comment makes me wonder whether there are any blockers to move to Ubuntu 22.04 LTS instead?

I propose to supersede this PR with #5651. Would that be fine with you, @camillem?

**Comment by sschuberth** [comment]
As #5651 is merged now which conflicts with this PR, I'll close this one.

**Comment by camillem** [comment]
> > This comment makes me wonder whether there are any blockers to move to Ubuntu 22.04 LTS instead?
> 
> I propose to supersede this PR with #5651. Would that be fine with you, @camillem?

Yes, of course. Thanks.

**Pull request closed without merging**